pkg-manager/core/test/install/overrides.ts

```diff
@@ -252,3 +250,47 @@ test('overrides with local file and link specs', async () => {
   expect(fs.realpathSync(path.join(indirectPrefix, '@pnpm.e2e/pkg-c'))).toBe(path.resolve('overrides/pkg'))
   expect(fs.realpathSync(path.join(indirectPrefix, '@pnpm.e2e/pkg-d'))).toBe(path.resolve('overrides/pkg'))
 })
+
+test('overrides remove dependencies', async () => {
+  const manifest: ProjectManifest = {
+    dependencies: {
+      '@pnpm.e2e/pkg-with-good-optional': '1.0.0',
+    },
+    pnpm: {
+      overrides: {
+        '@pnpm.e2e/pkg-with-good-optional>is-positive': '-',
+      },
+    },
+  }
+
+  const project = prepare(manifest)
+
+  await mutateModulesInSingleProject({
+    manifest,
+    mutation: 'install',
+    rootDir: process.cwd() as ProjectRootDir,
+  }, testDefaults({
+    overrides: manifest.pnpm?.overrides,
+  }))
+
+  // assert that @pnpm.e2e/pkg-with-good-optional@1.0.0 depends on is-positive@1.0.0
+  expect(project.requireModule('@pnpm.e2e/pkg-with-good-optional/package.json')).toMatchObject({
+    version: '1.0.0',
+    optionalDependencies: {
+      'is-positive': '1.0.0',
+    },
+  })
+
+  // yet because of the overrides, it installs @pnpm.e2e/pkg-with-good-optional@1.0.0 without is-positive@1.0.0
+  const lockfile = project.readLockfile()
+  expect(lockfile.snapshots).toHaveProperty(['@pnpm.e2e/pkg-with-good-optional@1.0.0'])
+  expect(lockfile.snapshots['@pnpm.e2e/pkg-with-good-optional@1.0.0']).not.toHaveProperty(['optionalDependencies', 'is-positive'])
+  expect(lockfile.snapshots).not.toHaveProperty(['optionalDependencies', 'is-positive'])
```

Comment:
I don't think this assertion makes sense.

Reply:
The snapshots assertions? It's to assert that the parent dep exist but doesn't have `is-positive` as a dependency.

Or did you mean that it was redundant?

Reply:
`lockfile.snapshots` won't ever have an `optionalDependencies` field.

Reply:
> `lockfile.snapshots` won't ever have an `optionalDependencies` field.

It does have an `optionalDependencies` field, at least in this case:

```diff
diff --git a/pkg-manager/core/test/install/overrides.ts b/pkg-manager/core/test/install/overrides.ts
index 63f2a1dcd..972d579b1 100644
--- a/pkg-manager/core/test/install/overrides.ts
+++ b/pkg-manager/core/test/install/overrides.ts
@@ -258,7 +258,7 @@ test('overrides remove dependencies', async () => {
     },
     pnpm: {
       overrides: {
-        '@pnpm.e2e/pkg-with-good-optional>is-positive': '-',
+        // '@pnpm.e2e/pkg-with-good-optional>is-positive': '-',
       },
     },
   }
@@ -284,6 +284,7 @@ test('overrides remove dependencies', async () => {
   // yet because of the overrides, it installs @pnpm.e2e/pkg-with-good-optional@1.0.0 without is-positive@1.0.0
   const lockfile = project.readLockfile()
   expect(lockfile.snapshots).toHaveProperty(['@pnpm.e2e/pkg-with-good-optional@1.0.0'])
+  console.debug(lockfile.snapshots['@pnpm.e2e/pkg-with-good-optional@1.0.0'])
   expect(lockfile.snapshots['@pnpm.e2e/pkg-with-good-optional@1.0.0']).not.toHaveProperty(['optionalDependencies', 'is-positive'])
   expect(lockfile.snapshots).not.toHaveProperty(['optionalDependencies', 'is-positive'])
   expect(lockfile.packages).toHaveProperty(['@pnpm.e2e/pkg-with-good-optional@1.0.0'])
```

![test screenshot](https://github.com/user-attachments/assets/20c5f0be-a755-4012-9709-c2c0904df276)

Reply:
But not directly as in `expect(lockfile.snapshots).not.toHaveProperty(['optionalDependencies', 'is-positive'])`

Reply:
Oh. I failed to notice the line below.

Reply:
[I have corrected it](https://github.com/pnpm/pnpm/pull/8576/commits/c432ef15ea993591a98727166e89a8bc073aaf79)